### ISSUE TITLE
2 Checks Recommended by Timesys Hardening Guide CONFIG_DEBUG_BUGVERBOSE & CONFIG_ZSMALLOC

### ISSUE THREAD

**Issue by wryMitts** [a13xp0p0v/kernel-hardening-checker#171]
Found here: 
https://timesys.com/pdf/Timesys-kernel-hardening-guide.pdf Dated 2022
https://web.archive.org/web/20241227042743/https://timesys.com/pdf/Timesys-kernel-hardening-guide.pdf

This seems useful for the project:
`DEBUG_BUGVERBOSE=is not set`

> Make sure this is not enabled, as it could provide an attacker
> sensitive kernel backtrace information on BUG() conditions

Not sure on this one?:
`ZSMALLOC=is not set`

> Do not enable memory allocator for compressed pages (slab-based
> memory allocator designed to store compressed RAM pages via
> virtual memory mapping

**Comment by a13xp0p0v** [comment]
@wryMitts, thanks for sharing this.

Looks like this **Timesys Corporation** did amazing job [SARCASM] selling their **Timesys Kernel Hardening Analysis Tool**, which simply provides the recommendations from `kernel-hardening-checker`.

They don't even mention `kernel-hardening-checker`. Shame on them:
https://www.timesys.com/security/securing-your-linux-configuration-kernel-hardening/

![Image](https://github.com/user-attachments/assets/a77f97cb-3646-460b-a574-64dd89702430)

**Comment by a13xp0p0v** [comment]
I agree with @thestinger, their rationale behind disabling `ZSMALLOC` is not very convincing.

I'm also not sure about disabling `DEBUG_BUGVERBOSE`.
All Linux distros rely on this. Maybe it's better to restrict unprivileged access to the kernel log (`CONFIG_SECURITY_DMESG_RESTRICT` and the `kernel.dmesg_restrict` sysctl)?

**Comment by jvoisin** [comment]
> I'm also not sure about disabling DEBUG_BUGVERBOSE.
All Linux distros rely on this. Maybe it's better to restrict unprivileged access to the kernel log (CONFIG_SECURITY_DMESG_RESTRICT and the kernel.dmesg_restrict sysctl)?

I agree, it's better to have the admin have access to verbose logs.

**Comment by a13xp0p0v** [comment]
> Maybe there could be an Android mode which takes this into account instead of it recommending enabling things which will be incompatible 

@thestinger, we have already considered having some special recommendations for Android in `kernel-hardening-checker`: https://github.com/a13xp0p0v/kernel-hardening-checker/pull/91

However, I don't know how to distinguish between kernel configs for GNU/Linux and Android: https://github.com/a13xp0p0v/kernel-hardening-checker/pull/142#issuecomment-2206492055

Do you have any suggestions?

**Comment by a13xp0p0v** [comment]
@thestinger, thanks for sharing your thoughts.

There is also another more general idea about how to enable customization of the checks: https://github.com/a13xp0p0v/kernel-hardening-checker/issues/50

Thinking about the proper design of this feature.
